src/manifold/src/manifold.cpp

```diff
@@ -767,4 +830,104 @@ Manifold Manifold::TrimByPlane(glm::vec3 normal, float originOffset) const {
 }
 
 ExecutionParams& ManifoldParams() { return params; }
+
+Manifold Manifold::Hull(const std::vector<float>& vertProps, const int numProp,
+                        const double precision) {
+  const int len = vertProps.size() / numProp;
+  // FIXME: Should there be a hull specific error?
+  if (len < 4) return Invalid();
+  std::vector<glm::vec3> pts(len);
+  for (int i = 0; i < len; i++) {
+    const int j = i * numProp;
+    pts[i] = {vertProps[j], vertProps[j + 1], vertProps[j + 2]};
+  }
+  const auto trip = NonCollinearTriple(pts, precision);
+  const auto plane = PlaneOfPts(pts[trip.x], pts[trip.y], pts[trip.z]);
+  const auto [d, dIsAboveTrip] = NonCoplanar(pts, plane, precision);
+  if (d < 0) return Invalid();  // All points coplanar.
+  const int a = trip.x;
+  const int b = dIsAboveTrip ? trip.z : trip.y;
+  const int c = dIsAboveTrip ? trip.y : trip.z;
+  std::vector<glm::ivec3> triangles;
+  std::vector<glm::vec4> planes;
+  std::unordered_set<glm::ivec2, IVec2Hash> halfEdges{};
+  std::vector<int> dropped;
+  std::vector<bool> kept;
+  const auto AddTri = [&pts, &triangles, &planes, &dropped, &kept](int a, int b,
+                                                                   int c) {
+    auto tri = glm::ivec3(c, b, a);
+    auto plane = PlaneOfPts(pts[a], pts[b], pts[c]);
+    if (dropped.size() > 0) {
+      const int idx = dropped[dropped.size() - 1];
+      triangles[idx] = tri;
+      planes[idx] = plane;
+      kept[idx] = true;
+      dropped.pop_back();
+    } else {
+      triangles.push_back(tri);
+      planes.push_back(plane);
+      kept.push_back(true);
+    }
+  };
+  AddTri(a, b, c);
+  AddTri(d, b, a);
+  AddTri(c, d, a);
+  AddTri(b, d, c);
+  for (int i = 0; i < len; i++) {
+    if (i == a || i == b || i == c || i == d) continue;  // skip starting points
```

Comment:
Maybe just swap these to the front of the vector and start at `i=4`?

Reply:
Of course, starting at 4 makes sense. That's a holdover from the way I was folding in the OCaml version.